tests/Functional/ContentEventRenderTest.php

```diff
@@ -183,42 +167,511 @@ public function testEventRegistrationDateDescription(): void {
+      'label' => 'Website',
+      'body' => 'Event website',
+    ];
+    $field_list_assert->assertPattern($field_list_expected_values, $practical_list_content->getOuterHtml());
+
+    // Assert "Registration capacity" field.
+    $node->set('oe_event_registration_capacity', 'event registration capacity')->save();
+    $this->drupalGet($node->toUrl());
+
+    $field_list_expected_values['items'][8] = [
+      'label' => 'Number of seats',
+      'body' => 'event registration capacity',
+    ];
+    $field_list_assert->assertPattern($field_list_expected_values, $practical_list_content->getOuterHtml());
+
+    // Assert "Entrance fee" field.
+    $node->set('oe_event_entrance_fee', 'entrance fee')->save();
+    $this->drupalGet($node->toUrl());
+
+    $field_list_expected_values['items'][9] = [
+      'label' => 'Entrance fee',
+      'body' => 'entrance fee',
+    ];
+    $field_list_assert->assertPattern($field_list_expected_values, $practical_list_content->getOuterHtml());
+
+    // Assert "Registration URL" field.
+    $node->set('oe_event_registration_url', 'http://www.example.com/registation');
+    $node->save();
+    $this->drupalGet($node->toUrl());
+
+    $registration_content = $this->assertSession()->elementExists('css', '#event-registration-block');
+    $this->assertRegisterButtonEnabled($registration_content);
+
+    // Assert "Registration date" field in the past.
+    $registration_start_date = (clone $static_time)->modify('- 2 months');
+    $registration_end_date = (clone $static_time)->modify('- 1 months');
+    $node->set('oe_event_registration_dates', [
+      'value' => $registration_start_date->format(DateTimeItemInterface::DATETIME_STORAGE_FORMAT),
+      'end_value' => $registration_end_date->format(DateTimeItemInterface::DATETIME_STORAGE_FORMAT),
+    ])->save();
+    $this->drupalGet($node->toUrl());
+
+    $this->assertSession()->elementNotExists('css', 'a.ecl-u-mt-2xl.ecl-link.ecl-link--cta', $registration_content);
     $registration_info_content = $this->assertSession()->elementExists('css', 'p.ecl-u-type-paragraph.ecl-u-type-color-grey-75');
-    $this->assertEquals('Registration will open today, 2 February 2020, 08:00.', $registration_info_content->getText());
+    $this->assertEquals('Registration period ended on Saturday 18 January 2020, 01:00', $registration_info_content->getText());
 
-    $this->setTimezone('America/New_York');
-    \Drupal::entityTypeManager()->getViewBuilder('node')->resetCache([$node]);
+    // Assert "Registration date" field when registration is in progress.
+    $registration_start_date = (clone $static_time)->modify('- 10 days');
+    $registration_end_date = (clone $static_time)->modify('+ 10 days');
+    $node->set('oe_event_registration_dates', [
+      'value' => $registration_start_date->format(DateTimeItemInterface::DATETIME_STORAGE_FORMAT),
+      'end_value' => $registration_end_date->format(DateTimeItemInterface::DATETIME_STORAGE_FORMAT),
+    ])->save();
```

Comment:
See above, when testing time-related features we must never change the actual event dates, we should change time, run cron and reload the page, instead.